cli/ss2

```diff
@@ -35,12 +38,82 @@ from pyroute2.netlink.diag import (SS_ESTABLISHED,
                                    SS_CONN)
 
 
-class Protocol(Callable):
+class UserCtxtMap(collections.Mapping):
+
+    _data = {}
+
+    _sk_inode_re = re.compile(r"socket:\[(?P<ino>\d+)\]")
+
+    _proc_sk_fd_cast = "/proc/%d/fd/%d"
+
+    def _parse_inode(self, pid, pconn):
+        sk_path = self._proc_sk_fd_cast % (pid, pconn.fd)
+        inode = None
+
+        sk_inode_raw = os.readlink(sk_path)
+        inode = self._sk_inode_re.search(sk_inode_raw).group('ino')
+
+        if not inode:
+            raise RuntimeError("Unexpected kernel sk inode outline")
+
+        return inode
+
+    def _enter_item(self, usr, pid, flow, ctxt):
+        sk_inode = int(self._parse_inode(pid, flow))
+        sk_fd = flow.fd
+
+        _enter_pid = lambda _: \
+            self._data[sk_inode][usr].__setitem__(pid, ctxt)
+
+        _enter_fd = lambda _: \
+            self._data[sk_inode][usr][pid]["fds"].append(sk_fd)
+
+        if sk_inode in self._data.keys():
+            if usr in self._data[sk_inode].keys():
+                if pid not in self._data[sk_inode][usr].keys():
+                    _enter_pid(ctxt)
+                _enter_fd(sk_fd)
+            else:
+                self._data[sk_inode][usr] = {}
+                _enter_pid(ctxt)
+                _enter_fd(sk_fd)
+        else:
+            self._data[sk_inode] = {}
+            self._data[sk_inode][usr] = {}
+            _enter_pid(ctxt)
+            _enter_fd(sk_fd)
```

Comment:
So if you believe the lambda usage is needed here (I don't), think about that:
1. If you need to call a lambda several times, one of the possible ways is to store lambdas in a dictionary. Technically it's the same as to define a class with methods. From the bytecode perspective it's almost the same.
2. Every variable used within the lambda should be passed as an argument, no implicit upper context lookups — simply 'cause it works a bit weird in some situations.


Here I would define two additional functions in some context, probably in the global context.